registry/handlers/app.go

```diff
@@ -211,44 +212,60 @@ func NewApp(ctx context.Context, config *configuration.Configuration) *App {
 		options = append(options, storage.EnableRedirect)
 	}
 
-	// configure storage caches
-	if cc, ok := config.Storage["cache"]; ok {
-		v, ok := cc["blobdescriptor"]
-		if !ok {
-			// Backwards compatible: "layerinfo" == "blobdescriptor"
-			v = cc["layerinfo"]
-		}
-
-		switch v {
-		case "redis":
-			if app.redis == nil {
-				panic("redis configuration required to use for layerinfo cache")
-			}
-			cacheProvider := rediscache.NewRedisBlobDescriptorCacheProvider(app.redis)
-			localOptions := append(options, storage.BlobDescriptorCacheProvider(cacheProvider))
-			app.registry, err = storage.NewRegistry(app, app.driver, localOptions...)
-			if err != nil {
-				panic("could not create registry: " + err.Error())
-			}
-			ctxu.GetLogger(app).Infof("using redis blob descriptor cache")
+	if config.Metadata.Enabled {
```

Comment:
In stead of having this be enabled/disabled, would it possible to define a unified metadata storage interface that defaults to using the existing storage drivers. The transactional and consistent access interfaces could be asserted where needed so we can have code comments and warnings about possible inconsistencies.